diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 1916aa2dc5..e6682849fa 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1019,6 +1019,10 @@ attribute_deprecated
 int64_t    av_stream_get_end_pts(const AVStream *st);
 #endif
 
+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff --git a/libavformat/mux_utils.c b/libavformat/mux_utils.c
index 3e63b8039a..1bc0a41770 100644
--- a/libavformat/mux_utils.c
+++ b/libavformat/mux_utils.c
@@ -40,6 +40,13 @@ int64_t av_stream_get_end_pts(const AVStream *st)
 }
 #endif
 
+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+    return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 int avformat_query_codec(const AVOutputFormat *ofmt, enum AVCodecID codec_id,
                          int std_compliance)
 {
