diff --git a/src/adw-bidi.c b/src/adw-bidi.c
index 76d50be..0991f0b 100644
--- a/src/adw-bidi.c
+++ b/src/adw-bidi.c
@@ -26,22 +26,9 @@
 
 #include "adw-bidi-private.h"
 
-#include <fribidi.h>
-
 static PangoDirection
 adw_unichar_direction (gunichar ch)
 {
-  FriBidiCharType fribidi_ch_type;
-
-  G_STATIC_ASSERT (sizeof (FriBidiChar) == sizeof (gunichar));
-
-  fribidi_ch_type = fribidi_get_bidi_type (ch);
-
-  if (!FRIBIDI_IS_STRONG (fribidi_ch_type))
-    return PANGO_DIRECTION_NEUTRAL;
-  else if (FRIBIDI_IS_RTL (fribidi_ch_type))
-    return PANGO_DIRECTION_RTL;
-  else
     return PANGO_DIRECTION_LTR;
 }
 
diff --git a/src/meson.build b/src/meson.build
index aa71da2..66cbaf0 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -293,7 +293,6 @@ appstream_dep = dependency('appstream',
 
 libadwaita_deps = [
   dependency('glib-2.0', version: glib_min_version),
-  dependency('fribidi'),
   gio_dep,
   gtk_dep,
   appstream_dep,
