diff --git a/meson.build b/meson.build
index 529669f..984d257 100644
--- a/meson.build
+++ b/meson.build
@@ -1694,16 +1694,6 @@ if with_llvm
     # without issues, so only apply this for other compilers.
     if dep_llvm.type_name() == 'internal'
       _llvm_rtti = subproject('llvm').get_variable('has_rtti', true)
-    else
-      # The CMake finder will return 'ON', the llvm-config will return 'YES'
-      _llvm_rtti = ['ON', 'YES'].contains(dep_llvm.get_variable(cmake : 'LLVM_ENABLE_RTTI', configtool: 'has-rtti'))
-    endif
-    if _rtti != _llvm_rtti
-      if _llvm_rtti
-        error('LLVM was built with RTTI, cannot build Mesa with RTTI disabled. Remove cpp_rtti disable switch or use LLVM built without LLVM_ENABLE_RTTI.')
-      else
-        error('LLVM was built without RTTI, so Mesa must also disable RTTI. Use an LLVM built with LLVM_ENABLE_RTTI or add cpp_rtti=false.')
-      endif
     endif
   endif
 
