diff --git a/platform/js/src/jit/x64/BaseAssembler-x641.h b/platform/js/src/jit/x64/BaseAssembler-x64.h
index bf9ca9f..9d78a11 100644
--- a/platform/js/src/jit/x64/BaseAssembler-x641.h
+++ b/platform/js/src/jit/x64/BaseAssembler-x64.h
@@ -592,6 +592,8 @@ class BaseAssemblerX64 : public BaseAssembler
 
     void movq_mr(int32_t offset, RegisterID base, RegisterID index, int scale, RegisterID dst)
     {
+    	if (!GPReg64Name(index))
+		return;	
         spew("movq       " MEM_obs ", %s", ADDR_obs(offset, base, index, scale), GPReg64Name(dst));
         m_formatter.oneByteOp64(OP_MOV_GvEv, offset, base, index, scale, dst);
     }
