diff --git a/third_party/jinja2/tests.py b/third_party/jinja2/tests.py
index 0adc3d4..c248367 100644
--- a/third_party/jinja2/tests.py
+++ b/third_party/jinja2/tests.py
@@ -10,7 +10,12 @@
 """
 import operator
 import re
-from collections import Mapping
+import sys
+if sys.version_info[:2] >= (3, 8):  # pragma: no cover
+    from collections.abc import Mapping
+else:  # pragma: no cover
+    from collections import Mapping
+
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types, integer_types
 import decimal
