diff --git a/tools/gn/build/gen.py b/tools/gn/build/gen.py
index dbf6035..57cacda 100755
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -442,9 +442,6 @@ def WriteGNNinja(path, platform, host, options, args_list):
       cflags.append('-fsanitize=undefined')
       ldflags.append('-fsanitize=undefined')
 
-    if not options.allow_warnings:
-      cflags.append('-Werror')
-
     cflags.extend([
         '-D_FILE_OFFSET_BITS=64',
         '-D__STDC_CONSTANT_MACROS', '-D__STDC_FORMAT_MACROS',
