diff --git a/extensions/renderer/bindings/argument_spec.h b/extensions/renderer/bindings/argument_spec.h
index 348bfb6..f6de11f 100644
--- a/extensions/renderer/bindings/argument_spec.h
+++ b/extensions/renderer/bindings/argument_spec.h
@@ -5,6 +5,7 @@
 #ifndef EXTENSIONS_RENDERER_BINDINGS_ARGUMENT_SPEC_H_
 #define EXTENSIONS_RENDERER_BINDINGS_ARGUMENT_SPEC_H_
 
+#include <cmath>
 #include <map>
 #include <memory>
 #include <set>
